pkg/websocket/server.go

```diff
@@ -107,28 +107,62 @@ func listenToWSServer(ctx context.Context, conn *gwebsocket.Conn) error {
 			continue
 		}
 
+		// ensure the environment is set up correctly
+		os.Setenv("KUBECONFIG", runtimeconfig.PathToKubeConfig())
+		os.Setenv("TMPDIR", runtimeconfig.EmbeddedClusterTmpSubDir())
+
 		switch msg.Command {
+		case "upgrade-manager":
+			d := map[string]string{}
+			if err := json.Unmarshal([]byte(msg.Data), &d); err != nil {
+				logrus.Errorf("failed to unmarshal data: %s: %s", err, string(msg.Data))
+				continue
+			}
+
+			reportStepStarted(ctx, kcli, d)
+
+			// path to the manager binary on the host
+			binPath := runtimeconfig.PathToEmbeddedClusterBinary("manager")
+
+			// TODO (@salah): airgap
+			err := manager.DownloadBinaryOnline(ctx, binPath, d["licenseID"], d["licenseEndpoint"], d["versionLabel"])
+			if err != nil {
+				reportStepFailed(ctx, kcli, d, fmt.Sprintf("failed to download manager binary: %s", err))
+				continue
+			}
+
+			// this is hacky but app slug is what determines the service name
```

Comment:
why not just exit?

Reply:
not sure i follow. why exit?

Reply:
> Exiting will exit the new process and start with the new binary

Per async discussion im ok with the systemctl restart